src/prognostic_equations/advection.jl

```diff
@@ -370,58 +369,97 @@ function edmfx_sgs_vertical_advection_tendency!(
     end
 
     for j in 1:n
-        @. ᶜa_scalar = draft_area(Y.c.sgsʲs.:($$j).ρa, ᶜρʲs.:($$j))
-        vtt = vertical_transport(
-            ᶜρʲs.:($j),
-            ᶠu³ʲs.:($j),
-            ᶜa_scalar,
-            dt,
-            edmfx_upwinding,
-        )
+        ᶜa = (@. lazy(draft_area(Y.c.sgsʲs.:($$j).ρa, ᶜρʲs.:($$j))))
+        edmf_upwnd = edmfx_upwinding
+
+        # Flux form vertical advection of area farction with the grid mean velocity
+        vtt = vertical_transport(ᶜρʲs.:($j), ᶠu³ʲs.:($j), ᶜa, dt, edmf_upwnd)
         @. Yₜ.c.sgsʲs.:($$j).ρa += vtt
 
-        va = vertical_advection(
-            ᶠu³ʲs.:($j),
-            Y.c.sgsʲs.:($j).mse,
-            edmfx_upwinding,
-        )
-        @. Yₜ.c.sgsʲs.:($$j).mse += va
+        # Advective form advection of mse and q_tot with the grid mean velocity
+        sgs_q_tot_mse = (@name(c.sgsʲs.:(1).mse), @name(c.sgsʲs.:(1).q_tot))
+        MatrixFields.unrolled_foreach(sgs_q_tot_mse) do qʲ_name
```

Comment:
I prefer something like χʲ_name instead of qʲ_name so that I know it includes mse.